index-scheduler/src/lib.rs

```diff
@@ -422,6 +430,7 @@ impl IndexScheduler {
                     ) {
                         std::thread::sleep(Duration::from_secs(1));
                     }
+                    run.wake_up.signal();
```

Comment:
Why do we have to signal the loop, here? Is it a bug fix?

Reply:
Yes it is a bug fix, but I didn't remember leaving it there. Basically we want the index scheduler to keep making progress even when a grave error occurs. Do you want it in a separate PR?

Reply:
No, that's ok but can you add it in the PR description please?